packages-exp/analytics-exp/testing/integration-tests/integration.ts

```diff
@@ -36,36 +37,71 @@ const RETRY_INTERVAL = 1000;
-  it('logEvent() sends correct network request.', async () => {
-    app = initializeApp(config);
-    logEvent(getAnalytics(app), 'login', { method: 'email' });
-    async function checkForEventCalls(): Promise<number> {
-      await new Promise(resolve => setTimeout(resolve, RETRY_INTERVAL));
-      const resources = performance.getEntriesByType('resource');
-      const callsWithEvent = resources.filter(
-        resource =>
-          resource.name.includes('google-analytics.com') &&
-          resource.name.includes('en=login')
-      );
-      if (callsWithEvent.length === 0) {
-        return checkForEventCalls();
-      } else {
-        return callsWithEvent.length;
+  describe('Using getAnalytics()', () => {
+    afterEach(() => deleteApp(app));
+    it('logEvent() sends correct network request.', async () => {
+      app = initializeApp(config);
+      logEvent(getAnalytics(app), 'login', { method: 'email' });
+      async function checkForEventCalls(): Promise<number> {
+        await new Promise(resolve => setTimeout(resolve, RETRY_INTERVAL));
+        const resources = performance.getEntriesByType('resource');
+        const callsWithEvent = resources.filter(
+          resource =>
+            resource.name.includes('google-analytics.com') &&
+            resource.name.includes('en=login')
+        );
+        if (callsWithEvent.length === 0) {
+          return checkForEventCalls();
+        } else {
+          return callsWithEvent.length;
+        }
       }
-    }
-    const eventCallCount = await checkForEventCalls();
-    expect(eventCallCount).to.equal(1);
+      const eventCallCount = await checkForEventCalls();
+      expect(eventCallCount).to.equal(1);
+    });
+    it("Warns if measurement ID doesn't match.", done => {
+      const warnStub = stub(console, 'warn').callsFake(() => {
+        expect(warnStub.args[0][1]).to.include('does not match');
+        done();
+      });
+      app = initializeApp({
+        ...config,
+        measurementId: 'wrong-id'
+      });
+      getAnalytics(app);
+    });
   });
-  it("Warns if measurement ID doesn't match.", done => {
-    const warnStub = stub(console, 'warn').callsFake(() => {
-      expect(warnStub.args[0][1]).to.include('does not match');
-      done();
+  describe('Using initializeAnalytics()', () => {
+    it('logEvent() sends correct network request.', async () => {
+      app = initializeApp(config);
+      logEvent(initializeAnalytics(app), 'login', { method: 'email' });
+      async function checkForEventCalls(): Promise<PerformanceEntry[]> {
+        await new Promise(resolve => setTimeout(resolve, RETRY_INTERVAL));
+        const resources = performance.getEntriesByType('resource');
+        const callsWithEvent = resources.filter(
+          resource =>
+            resource.name.includes('google-analytics.com') &&
+            resource.name.includes('en=login')
+        );
+        if (callsWithEvent.length === 0) {
+          return checkForEventCalls();
```

Comment:
We should add a limit on the number of retries, otherwise it may run forever.

Reply:
I thought the karma timeout will catch it but I added a timeout and extracted the function, which helped me realize the second test was not correct (performance entries were not cleared from the browser between tests), so I fixed it.